### ISSUE TITLE
Docs for API migration from NuGet 2.x to NuGet 4+

### ISSUE THREAD

**Issue by Rick-van-Dam** [NuGet/Home#8312]
## Details about Problem

Its currently not clear how one can integrate nuget in a custom application. For instance [duality](https://www.duality2d.net/) has its own package manager that uses nuget (currently 2.x) internally. However in this version we are running into issues with netstandard where it does not install any dependencies for netstandard packages causing runtime exceptions. We tried [upgrading](https://github.com/ilexp/NuGet4xTest) to nuget 4.x but this didn't work out as its not clear how to use the API.

Please provide documentation on how to implement nuget in a custom application.

**Comment by nkolev92** [comment]
Unfortunately we don't have resources diving into scenarios like yours. 

There are some external resources: 
https://docs.microsoft.com/en-us/nuget/reference/nuget-client-sdk

Another good resource is the source code https://github.com/NuGet/NuGet.Client/. 

The reason for your issues is likely the old NuGet APIs not understanding .NET Standard.

Can you describe in more detail what you are trying to do, and maybe I can give you some pointers.

**Comment by Rick-van-Dam** [comment]
Duality has its own package manager thats used to install so called plugins. These plugins are currently PCL's but we want to update these to netstandard. We did some work on this but I noticed that nuget 2.14 does not install any dependencies when you install a package that targets only netstandard2.0 in a netframework 4.7.2 application.

This seems to be pretty much blocking us from going to netstandard.

The new nuget api has changed so drastically that it becomes really hard to update to 4.x
